stl/inc/functional

```diff
@@ -1965,16 +2064,20 @@ public:
         using _Vt = decay_t<_Fn>;
         static_assert(is_constructible_v<_Vt, _Fn>, "_Vt should be constructible from _Fn. "
                                                     "(N4950 [func.wrap.move.ctor]/6)");
+        using _VtInvQuals = _Call::template _VtInvQuals<_Vt>;
 
-        if constexpr (is_member_pointer_v<_Vt> || is_pointer_v<_Vt> || _Is_specialization_v<_Vt, move_only_function>) {
-            if (_Callable == nullptr) {
-                this->_Reset_to_null();
-                return;
+        if constexpr (is_same_v<_Vt, function<_Signature...>>) {

```

Comment:
This optimization requires exactly the same signature and it avoids an invoke call (with potential objects copying) and often an allocation.

There's very similar but different optimization possible that would accept different but compatible signature, and will avoid an allocation, but not a call.

One thing at a time.

Reply:
Starting with matching signatures is definitely good. I'm not sure if handling varying signatures is worth additional implementation complexity, but I could be persuaded.

Reply:
Not doing different signatures here per #5862 resolution